torch/jit/_recursive.py

```diff
@@ -559,6 +559,13 @@ def check_module_initialized(mod):
         raise RuntimeError("'{}' has not been initialized, did you forget to call 'super()'?"
                            .format(torch.typename(type(mod))))
 
+    # This is to avoid importing torch.distributed.nn
+    if not hasattr(mod, 'remote_parameters'):
+        for param in mod.parameters():
```

Comment:
This does not help for the internal tests as you're still accessing `.parameters()` on a Module-like object that don't have such a function.
As mentioned before, I think you need to iterate over the `._parameters` directly here.